scripts/qa_crowbarsetup.sh

```diff
@@ -4644,6 +4644,9 @@ function onadmin_testsetup
     if iscloudver 7plus && [[ $cloudsource =~ 'develcloud' ]] ; then
         test_keystone_toggle_ssl
         keystoneret=$?
+
+        # Make sure everything is in sync before starting tempest
+        crowbar_proposal_commit provisioner
     fi
 
     oncontroller testsetup
```

Comment:
The crm failcount still happens in this step, that's why the toggle test originally happened after this.

Why can we not just move the toggle back to after this step?

Reply:
@cmurphy the original reason for me was that for selfish reasons it was difficult to find the actual testsetup report "e.g. which part failed" because it was way outside the short log. 

we can consider reverting my change for the time being if this patch has no chance of getting reviewed in time.

Reply:
I'm missing some context here.  Do we know why toggling SSL causes failures, what those failures are, and that it would be significantly harder to fix them than to include this `crm resource cleanup` hack?  If so, presumably we'd also need to document this hack too?  If not, is it realistic to do a quick investigation?  Not sure what you mean by "short log" here?

Reply:
@aspiers turning SSL on or off means that the catalog needs to be updated, OpenStack services needs to switch to using the new endpoint and be reloaded, Apache needs to be updated and reloaded, and haproxy needs to be updated and reloaded. That takes several seconds, but even if it was instantaneous, all clients need to switch to a new endpoint. It would be impossible to do this transparently. So checking the failcounts after this toggling is done is pointless because it's known to be a disruptive operation.

Reply:
@cmurphy Thank you, that is *extremely* helpful.  I wish it had been in the commit messages or at least in the PR description.  Now that I understand, I think the technically correct way to handle this toggling would be to put everything in maintenance mode before-hand, and take it out afterwards.  AFAIK that's exactly the kind of scenario which maintenance mode was intended for.  Can we try that instead of this ugly `crm resource cleanup` hack?  Lars pointed out to me several years ago that the regularity with which the SOC team uses `crm resource cleanup` was alarming and indicates that we are using Pacemaker in the wrong ways ...

Reply:
@aspiers updated the commit message with colleen's explanation.